### PR TITLE
feat(react-grid): support column virtualization

### PR DESCRIPTION
**Pull request by kvet**

### CODE REVIEW

File: packages/dx-react-core/src/forward-ref.jsx [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import * as React from 'react';
+import * as PropTypes from 'prop-types';
+
+export class ForwardRef extends React.PureComponent {
```
Comment (gsobolev) [inline packages/dx-react-core/src/forward-ref.jsx]: This component seems weird. Maybe we can rename it to something more definitive.

File: packages/dx-grid-core/src/plugins/table-group-row/helpers.js [hunk 2]
```diff
@@ -3,7 +3,13 @@ import { TABLE_GROUP_TYPE } from './constants';
 export const isGroupTableCell = (tableRow, tableColumn) =>
   tableRow.type === TABLE_GROUP_TYPE && tableColumn.type === TABLE_GROUP_TYPE
   && tableRow.row.groupedBy === tableColumn.column.name;
-export const isGroupIndentTableCell = (tableRow, tableColumn) =>
-  tableRow.type === TABLE_GROUP_TYPE && tableColumn.type === TABLE_GROUP_TYPE
-  && tableRow.row.groupedBy !== tableColumn.column.name;
+export const isGroupIndentTableCell = (tableRow, tableColumn, grouping) => {
+  if (tableRow.type !== TABLE_GROUP_TYPE || tableColumn.type !== TABLE_GROUP_TYPE) return false;
+  if (tableRow.row.groupedBy === tableColumn.column.name) return false;
```
[the remark this answers is not in the document]
Comment (kvet) [reply packages/dx-grid-core/src/plugins/table-group-row/helpers.js]: Yes, we can. But the row will be too long. We will need to split it.

File: packages/dx-react-grid/src/plugins/table.jsx [hunk 3]
```diff
@@ -57,13 +59,19 @@ export class Table extends React.PureComponent {
         <Getter name="tableHeaderRows" value={tableHeaderRows} />
         <Getter name="tableBodyRows" computed={tableBodyRowsComputed} />
         <Getter name="tableColumns" computed={tableColumnsComputed} />
+        <Getter name="getTableCellColSpan" value={tableCellColSpanGetter} />
```
[the remark this answers is not in the document]
Comment (kvet) [reply packages/dx-react-grid/src/plugins/table.jsx]: This is conceived. I am not sure that it is the final realization of the colSpan API. So, it's better to hide it from docs.

File: packages/dx-react-grid/src/components/table-layout/virtual-table-layout.jsx [hunk 4]
```diff
@@ -108,81 +92,136 @@ export class VirtualTableLayout extends React.PureComponent {
...
+      >
+        <ColumnGroup
+          columns={collapsedGrid.columns.map(({ key, column, width: columnWidth }) =>
+            ({ key: key || column.key, width: columnWidth }))}
+        />
+        <Body>
+          {collapsedGrid.rows.map((visibleRow) => {
+            if (visibleRow.type === 'stub') {
+              return (
+                <tr key={visibleRow.key} style={{ height: `${visibleRow.height}px` }} />
```
Comment (SergeyAlexeev) [inline packages/dx-react-grid/src/components/table-layout/virtual-table-layout.jsx]: I think we shouldn't use html tags(`tr`, `td`) in this package.

File: packages/dx-react-grid/src/components/table-layout/virtual-table-layout.jsx [hunk 5]
```diff
@@ -108,81 +92,136 @@ export class VirtualTableLayout extends React.PureComponent {
...
+                <Row
+                  tableRow={row}
+                  style={row.height !== undefined
+                    ? { height: `${row.height}px` }
+                    : undefined}
+                >
+                  {cells.map((cell) => {
+                    if (cell.type === 'stub') {
+                      return (
+                        <td
```
Comment (SergeyAlexeev) [inline packages/dx-react-grid/src/components/table-layout/virtual-table-layout.jsx]: The same here.

File: packages/dx-grid-core/src/plugins/table-row-detail/computeds.js [hunk 6]
```diff
@@ -30,3 +29,11 @@ export const tableColumnsWithDetail = (tableColumns, toggleColumnWidth) => [
   { key: TABLE_DETAIL_TYPE, type: TABLE_DETAIL_TYPE, width: toggleColumnWidth },
   ...tableColumns,
 ];
+
+export const tableDetailRowCellColSpanGetter = getTableCellColSpan => (params) => {
```
Comment (gsobolev) [inline packages/dx-grid-core/src/plugins/table-row-detail/computeds.js]: Can we make this name shorter, e.g. 'tableDetailCellColSpanGetter'?

**Pull request merged**